src/process_utility.c

```diff
@@ -778,23 +810,23 @@ process_reindex(Node *parsetree)
 	if (!OidIsValid(relid))
 		return false;
 
-	hcache = ts_hypertable_cache_pin();
-
 	switch (stmt->kind)
 	{
 		case REINDEX_OBJECT_TABLE:
-			ht = ts_hypertable_cache_get_entry(hcache, relid);
+			ht = ts_hypertable_cache_get_entry(args->hcache, relid);
 
 			if (NULL != ht)
 			{
 				PreventCommandDuringRecovery("REINDEX");
 
+				process_add_hypertable(args, ht);
+
 				if (foreach_chunk(ht, reindex_chunk, stmt) >= 0)
 					ret = true;
 			}
 			break;
 		case REINDEX_OBJECT_INDEX:
-			ht = ts_hypertable_cache_get_entry(hcache, IndexGetRelation(relid, true));
+			ht = ts_hypertable_cache_get_entry(args->hcache, IndexGetRelation(relid, true));
 
 			if (NULL != ht)
```

Comment:
no need to `process_add_hypertable` here? Even if this is an error now, it may not be in the future...